### PR TITLE
CI/CD: add inclavare containers development centos8.2 and ubuntu 18.04 docker images

### PR DESCRIPTION
**Pull request by hustliyilin**
Signed-off-by: Yilin Li YiLin.Li@linux.alibaba.com

### CODE REVIEW

**Comment by hustliyilin** [comment]
- build test: https://github.com/hustliyilin/inclavare-containers/actions/runs/610511798
- the build of rune、shim、 epm、sgx-tools、skeleton is ok.
- the docker is installed in the docker images.
- the rpm and deb build don't need to be installed any packages.

File: .github/workflows/docker/Dockerfile-dev-centos8.2 [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# install go
+RUN wget https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -zxvf go1.14.2.linux-amd64.tar.gz -C /usr/lib && \
+    rm -rf go1.14.2.linux-amd64.tar.gz
+
+# configure GOPATH and GOROOT
+ENV GOROOT       /usr/lib/go
+ENV GOPATH       /root/gopath
+ENV PATH         $PATH:$GOROOT/bin:$GOPATH/bin
+ENV GOPROXY      "https://mirrors.aliyun.com/goproxy,direct"
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-centos8.2:20]: Please remove this line. This base docker image is common to use for any developers. We can use multi-stage building to involve this line for CI/CD testing.

File: .github/workflows/docker/Dockerfile-dev-centos8.2 [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+WORKDIR /root
+
+# install go
+RUN wget https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -zxvf go1.14.2.linux-amd64.tar.gz -C /usr/lib && \
+    rm -rf go1.14.2.linux-amd64.tar.gz
+
+# configure GOPATH and GOROOT
+ENV GOROOT       /usr/lib/go
+ENV GOPATH       /root/gopath
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-centos8.2:18]: Plz create this folder later.

File: .github/workflows/docker/Dockerfile-dev-centos8.2 [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ENV PATH         $PATH:$GOROOT/bin:$GOPATH/bin
+ENV GOPROXY      "https://mirrors.aliyun.com/goproxy,direct"
+ENV GO111MODULE  on
+
+# install and configure docker
+RUN dnf --enablerepo=PowerTools install -y iptables && \
+    wget https://download.docker.com/linux/static/stable/x86_64/docker-18.09.7.tgz && \
+    tar -zxvf docker-18.09.7.tgz && mv docker/* /usr/bin
+
+RUN mkdir -p /etc/docker && \
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-centos8.2:28]: Add a comment line to explain what it is doing.

File: .github/workflows/docker/Dockerfile-dev-centos8.2 [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    echo -e "{\n\t\"runtimes\": {\n\t\t\"rune\": {\n\t\t\t\"path\": \"/usr/local/bin/rune\",\n\t\t\t\"runtimeArgs\": []\n\t\t}\n\t}\n}" >> /etc/docker/daemon.json
+
+# install dcap
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-centos8.2:31]: Not dcap. It is Intel SGX installation packages.

File: .github/workflows/docker/Dockerfile-dev-centos8.2 [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+RUN wget https://download.01.org/intel-sgx/sgx-linux/2.13/distro/centos8.2-server/sgx_rpm_local_repo.tgz && \
+    tar -zxvf sgx_rpm_local_repo.tgz && rm sgx_rpm_local_repo.tgz && cd sgx_rpm_local_repo && rpm -ivh libsgx-headers*.rpm && rpm -ivh libsgx-dcap-quote-verify*.rpm && cd ../ && rm -rf sgx_rpm_local_repo
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-centos8.2:33]: Just run `rpm -ivh libsgx-dcap-quote-verify-devel-*.rpm`.

File: .github/workflows/docker/Dockerfile-dev-ubuntu18.04 [hunk 6]
```diff
@@ -0,0 +1,34 @@
+FROM ubuntu:18.04
+
+LABEL maintainer="YiLin Li <YiLin.Li@linux.alibaba.com>"
+
+RUN apt-get update && apt-get install -y make wget git \
+    libseccomp-dev gnupg gcc pkg-config libssl-dev libprotobuf-c0-dev \
+    protobuf-c-compiler binutils-dev debhelper
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-ubuntu18.04:7]: protobuf provided by ubuntu 18.04 are not useful for us. We need to build protobuf-c from scratch by self.

In addition, these dependencies are not enough to build shelter from scratch.

File: .github/workflows/docker/Dockerfile-dev-ubuntu18.04 [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+WORKDIR /root
+
+# install go
+RUN wget https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -zxvf go1.14.2.linux-amd64.tar.gz -C /usr/lib && \
+    rm -rf go1.14.2.linux-amd64.tar.gz
+
+# configure GOPATH and GOROOT
+ENV GOROOT       /usr/lib/go
+ENV GOPATH       /root/gopath
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-ubuntu18.04:18]: create this folder later.

File: .github/workflows/docker/Dockerfile-dev-ubuntu18.04 [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ENV PATH         $PATH:$GOROOT/bin:$GOPATH/bin
+ENV GOPROXY      "https://mirrors.aliyun.com/goproxy,direct"
+ENV GO111MODULE  on
+
+# install and configure docker
+RUN apt-get install -y iptables && \
+    wget https://download.docker.com/linux/static/stable/x86_64/docker-18.09.7.tgz && \
+    tar -zxvf docker-18.09.7.tgz && mv docker/* /usr/bin
+
+RUN mkdir -p /etc/docker && \
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-ubuntu18.04:28]: Add a comment line.

File: .github/workflows/docker/Dockerfile-dev-ubuntu18.04 [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    echo "{\n\t\"runtimes\": {\n\t\t\"rune\": {\n\t\t\t\"path\": \"/usr/local/bin/rune\",\n\t\t\t\"runtimeArgs\": []\n\t\t}\n\t}\n}" >> /etc/docker/daemon.json
+
+# install dcap
+RUN echo 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main' | tee /etc/apt/sources.list.d/intel-sgx.list && wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add -
```
Comment (jiazhang0) [inline .github/workflows/docker/Dockerfile-dev-ubuntu18.04:32]: Use "\" to avoid signing long line.

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

**Comment by jiazhang0** [comment]
>     * build test: https://github.com/hustliyilin/inclavare-containers/actions/runs/610511798
> 
>     * the build of `rune`、`shim`、 `epm`、`sgx-tools`、`skeleton` is ok.
> 
>     * the docker is installed in the docker images.
> 
>     * the rpm and deb build don't need to be installed any packages.

shelter and ra-tls are not tested?

**Comment by jiazhang0** [comment]
This docker image is created for inclavare development, so it is necessary to make shelter and ra-tls buildable. In addition, the workflow names should be renamed as 'Dockerfile-inclavare-dev-foo'.

**Comment by hustliyilin** [comment]
> > ```
> > * build test: https://github.com/hustliyilin/inclavare-containers/actions/runs/610511798
> > 
> > * the build of `rune`、`shim`、 `epm`、`sgx-tools`、`skeleton` is ok.
> > 
> > * the docker is installed in the docker images.
> > 
> > * the rpm and deb build don't need to be installed any packages.
> > ```
> 
> shelter and ra-tls are not tested?



> > ```
> > * build test: https://github.com/hustliyilin/inclavare-containers/actions/runs/610511798
> > 
> > * the build of `rune`、`shim`、 `epm`、`sgx-tools`、`skeleton` is ok.
> > 
> > * the docker is installed in the docker images.
> > 
> > * the rpm and deb build don't need to be installed any packages.
> > ```
> 
> shelter and ra-tls are not tested?

Non-core components will be added later but not now.

**Pull request closed without merging**